GSEA KEGG Results
========================================================

We have compared the global gene expression profiles of the UISO and classic cell lines to the tumors and each other. At this level, we observe that the UISO cell line is not similar to the tumor samples nor the classic cell line, while the classic cell lines are similar to the tumor samples. We would also like to show that at the pathway level, the pathways that are differentially expressed between the UISO cell line and the tumor samples are not the same as those differentially expressed between the WaGa cell line and the tumor samples. Lastly, we will show that the pathways differentially expressed between the UISO cell line and the tumor are the same as those that are differentially expressed between the UISO cell line and the classic cell lines. 

To identify potential biological similarities and differences, we use a subset of pathways taken from the Kyoto Encyclopedia of Genes and Genomes (KEGG) database and apply the Gene Set Enrichment Analysis (GSEA) tool "GseaPreranked" to identify pathways that are differentially expressed between the cell lines and the tumor as well as between the two cell lines. As input, we used the moderated t-statistic from the 'limma' package for differential expression. When running GseaPreranked, we use the default weighted method (p = 1) for computing the running sum statistic. All other parameters were set as the defaults (see code for exact command line calls). When evaluating the results from GSEA, we use the default FDR threshold of 25% to identify significantly affected pathways.

```{r loadlibs, echo=FALSE, message=FALSE}
# Read GSEA results for KEGG and combine to a table
library(plyr)
library(xtable)
library(reshape)
library(pander)
library(ggplot2)
```

``` {r rungsea, echo=FALSE, message=FALSE, eval=FALSE}
## Commands to run GSEA using GSEAPreranked on the output from limma's moderated t-test

# Unfortunately, this step is difficult to automatically reproduce because GSEA writes out files to a folder with a timestamp appended.
# All files run with these parameters are provided.

# The following code will reproduce these files, but the timestamp of the output folder will be different, and you would need to change
# that in the following code manually.

# From the command line, assumes that your current directory is the repository root:
# > sh ./data/GSEA/run_GSEA.sh

```

```{r readdata, echo=FALSE}
data <- list()
data[['tumor.classic.pos']] <- read.table("../data/GSEA/tumor_vs_classic_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['tumor.classic.pos']]$direction <- 1

data[['tumor.classic.neg']] <- read.table("../data/GSEA/tumor_vs_classic_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['tumor.classic.neg']]$direction <- -1

data[['classic.variant.pos']] <- read.table("../data/GSEA/classic_vs_variant_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['classic.variant.pos']]$direction <- 1

data[['classic.variant.neg']] <- read.table("../data/GSEA/classic_vs_variant_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['classic.variant.neg']]$direction <- -1

data[['tumor.variant.pos']] <- read.table("../data/GSEA/tumor_vs_variant_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['tumor.variant.pos']]$direction <- 1

data[['tumor.variant.neg']] <- read.table("../data/GSEA/tumor_vs_variant_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['tumor.variant.neg']]$direction <- -1

merged.data <- ldply(data, function(x) x[, c("NAME", "FDR.q.val", "direction")])

## Process data
merged.data$NAME <- gsub("KEGG_", "", merged.data$NAME)
merged.data$.id <- gsub("\\.pos|\\.neg", "", merged.data$.id)
merged.data$FDR.q.val <- round(merged.data$FDR.q.val, 6)

# fdr.thresh.init <- 0.25
# merged.data <- subset(merged.data, FDR.q.val <= fdr.thresh.init)
# merged.data <- subset(merged.data, FDR.q.val <= 0.95 & FDR.q.val >= 0.9)

merged.data$FDR.q.val.formatted <- format.pval(merged.data$FDR.q.val, digits=4)

merged.data$value <- ifelse(merged.data$direction == 1,
                            gsub("(.*)", '<div style="color: red;">\\1</div>', merged.data$FDR.q.val),
                            gsub("(.*)", '<div style="color: green;">\\1</div>', merged.data$FDR.q.val))

fdr.thresh <- 0.05
merged.data$value <- ifelse(merged.data$FDR.q.val <= fdr.thresh,
                            gsub("(.*)", '<div style="font-size:150%">\\1</div>', merged.data$value),
                            merged.data$value)

  
merged.data.wide <- cast(merged.data, NAME ~ .id, value="value")
rownames(merged.data.wide) <- merged.data.wide$NAME
merged.data.wide <- merged.data.wide[, -1]
merged.data.wide <- as.data.frame(merged.data.wide)

merged.data.wide.fdr <- cast(merged.data, NAME ~ .id, value="FDR.q.val")
rownames(merged.data.wide.fdr) <- rownames(merged.data.wide)
merged.data.wide.fdr <- merged.data.wide.fdr[, -1]

merged.data.wide.fdr <- as.data.frame(apply(merged.data.wide.fdr, 2, as.numeric))
rownames(merged.data.wide.fdr) <- rownames(merged.data.wide)
colnames(merged.data.wide.fdr) <- colnames(merged.data.wide)

## Anything that is NA is FDR = 1
merged.data.wide.fdr[is.na(merged.data.wide.fdr)] <- 1

merged.data.wide <- merged.data.wide[order(merged.data.wide.fdr$tumor.variant), ]

merged.data.wide$classic.variant <- as.character(merged.data.wide$classic.variant)
merged.data.wide$tumor.variant <- as.character(merged.data.wide$tumor.variant)
merged.data.wide$tumor.classic <- as.character(merged.data.wide$tumor.classic)

set.na <- is.na(merged.data.wide.fdr)
merged.data.wide[set.na] <- "NS" # paste("> ", fdr.thresh, sep="")

merged.data.wide <- merged.data.wide[, c("tumor.variant", "tumor.classic", "classic.variant")]

colnames(merged.data.wide) <- c( "UISO vs. Tumor", "Classic vs. Tumor", "UISO vs. Classic")

```

## Results for GSEA, differentially expressed in the cell lines (number of pathways: `r nrow(merged.data.wide)`):

To identify pathways that are affected similarly, we performed three comparisons: UISO cell line vs. tumor samples, classic cell lines vs. tumor samples, and UISO cell line vs. classic cell lines. Green values are up-regulated and red values are down-regulated. Large bold values are statistically significant (FDR-adjusted p-value < `r fdr.thresh`).

```{r table.all, echo=FALSE, results='asis'} 
print(xtable(merged.data.wide, digits=c(1,2,2,2), align=c("r", "c", "c", "c")), type='html', sanitize.text.function=I) 
``` 
