Supplemental figures and tables
=================================

```{r load, message=FALSE, warning=FALSE, echo=FALSE}
# Here's how to compile this file in R, assuming your current directory is the repository root:

# library(knitr)
# knit("./reports/SuppMat.Rmd", encoding="utf-8")

library(affy)
library(ggplot2)
library(reshape)
library(plyr)
library(cluster)
library(xtable)
library(MASS)

options(xtable.type="html", xtable.caption.placement="top", xtable.include.row.names=FALSE)
```

Supplemental Figure 1: The WaGa and Mkl-1 cell lines are more similar to MCC tumor samples than the UISO cell line.
-------------------
Boxplots of pairwise Spearman correlations ($\rho$) of global, variance-filtered probeset expression signatures between different cell line samples (WaGa, Mkl-1 and UISO) and between cell line and tumor samples. For each box: middle bar, median; box, inter-quartile range (middle 50\% of data); bars extend to 1.5 times the inter-quartile range. The color of individual points indicates the MCV status of the tumor samples.

```{r loadexpr, echo=FALSE}
load("../cache/eset.original.filter.RData")
dataset <- eset.original.filter
```

```{r figS1, message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=7}

dataset.original <- eset.original.filter

# Subset of samples and make sample classes
which.MCCtumor <- grep("MT.*", sampleNames(dataset.original))
which.mkl1 <- grep("Mkl.*", sampleNames(dataset.original))
which.waga <- grep("Waga.*", sampleNames(dataset.original))
which.uiso <- grep("UISO.*", sampleNames(dataset.original))

keep <- c(which.MCCtumor, which.mkl1, which.waga, which.uiso)
dataset.original <- dataset.original[, keep]

dataset <- dataset.original

# Compute correlation
dataset.cor <- cor(exprs(dataset), method='spearman')

# Keep only the necessary columns and rows and
# set self correlation values (including within group) to NA
dataset.cor[upper.tri(dataset.cor, diag=TRUE)] <- NA
dataset.cor[grep("Mkl", rownames(dataset.cor)),
            grep("Mkl", rownames(dataset.cor))] <- NA
dataset.cor[grep("Waga", rownames(dataset.cor)),
            grep("Waga", rownames(dataset.cor))] <- NA
dataset.cor[grep("UISO", rownames(dataset.cor)),
            grep("UISO", rownames(dataset.cor))] <- NA

exclude.rows <- -(1:(length(which.MCCtumor)))
dataset.cor <- dataset.cor[exclude.rows, ]

# Reshape the data
dataset.cor <- transform(dataset.cor, sample=rownames(dataset.cor))
dataset.cor.melted <- melt(dataset.cor, id.vars=c("sample"), na.rm=TRUE)

# For some reason, melt changed the Mkl-1's name...
dataset.cor.melted$variable <- factor(gsub("\\.", "-", 
                                           dataset.cor.melted$variable))

# Get sample_class from the phenotypeData based on the sample column
dataset.cor.melted <- merge(dataset.cor.melted,
                            pData(dataset)[, c("sample", "class")],
                            by.x="sample", by.y="sample")

dataset.cor.melted$class <- factor(dataset.cor.melted$class, 
                                   levels=c("WaGa", "Mkl1", "UISO"),
                                   # labels=c("WaGa", "Mkl-1", "UISO"),
                                   ordered=TRUE)

# Get everything from the phenotypeData again, based on the variable column
dataset.cor.melted <- merge(dataset.cor.melted, pData(dataset.original), by.x="variable", by.y="sample")
dataset.cor.melted <- dataset.cor.melted[, c('sample', 'variable', 'value', 'class.x', "class.y", "MCPyV.status")]

# Add some grouping factors
dataset.cor.melted <- transform(dataset.cor.melted,
                                comparison=factor(with(dataset.cor.melted, paste(class.x, class.y, sep=".")),
                                                  levels=c("WaGa.Tumor", "Mkl1.Tumor", "UISO.Tumor", "WaGa.Mkl1", "UISO.WaGa", "UISO.Mkl1"),
                                                  labels=c("WaGa.Tumor", "Mkl-1.Tumor", "UISO.Tumor", "WaGa.Mkl-1", "UISO.WaGa", "UISO.Mkl-1"),
                                                  ordered=TRUE),
                                tumorORnot=factor(dataset.cor.melted$class.y == "Tumor", #c("MCC Tumor", "SCLC Tumor"), 
                                                  labels=c("Cell line vs. cell line", "Cell line vs. tumor")))


# plot
p <- ggplot(dataset.cor.melted, aes(x=comparison, y=value))
p <- p + geom_boxplot(notch=FALSE, outlier.size=0)
p <- p + geom_jitter(aes(color=MCPyV.status, shape=MCPyV.status), position=position_jitter(width=0.1), alpha=0.6)
p <- p + facet_grid(. ~ tumorORnot, scales="free_x")
p <- p + labs(x="", y=expression(Correlation~(rho)))
p <- p + scale_color_brewer(palette='Set1')
p <- p + guides(colour=guide_legend("Tumor MCV Status"), shape=guide_legend("Tumor MCV Status"))
p <- p + ylim(c(0.5,1))
p <- p + theme_bw() + theme(axis.text=element_text(size=20), 
                            legend.title=element_text(size=20),
                            legend.text=element_text(size=20),
                            legend.position=c(0.6, 0.2),
                            axis.title.x=element_text(size=20),
                            axis.title.x=element_text(size=20),
                            axis.title.y=element_text(size=20, angle=90),
                            strip.text.x=element_text(size=24))

print(p)

ggsave("../graphs/figS1.pdf", p, width=16, height=8)

# # to see the median values:
# ddply(dataset.cor.melted, .(comparison), function(x) data.frame(median=median(x$value)))

```                        

```{r mcvcorr, echo=FALSE, results='asis', include=FALSE}
# Show the median correlation values by mcv status
mcv.corr <- ddply(dataset.cor.melted, .(comparison, MCPyV.status), function(x) data.frame(median=median(x$value)))
mcv.corr.cast <- cast(mcv.corr, comparison ~ MCPyV.status, value="median")
xtable(mcv.corr.cast)
```

```{r readdata, echo=FALSE}
data <- list()
data[['tumor.classic.pos']] <- read.table("../data/GSEA/tumor_vs_classic_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['tumor.classic.pos']]$direction <- 1

data[['tumor.classic.neg']] <- read.table("../data/GSEA/tumor_vs_classic_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['tumor.classic.neg']]$direction <- -1

data[['classic.variant.pos']] <- read.table("../data/GSEA/classic_vs_variant_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['classic.variant.pos']]$direction <- 1

data[['classic.variant.neg']] <- read.table("../data/GSEA/classic_vs_variant_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['classic.variant.neg']]$direction <- -1

data[['tumor.variant.pos']] <- read.table("../data/GSEA/tumor_vs_variant_filter/gsea_report_for_na_pos.xls", sep="\t", header=TRUE)
data[['tumor.variant.pos']]$direction <- 1

data[['tumor.variant.neg']] <- read.table("../data/GSEA/tumor_vs_variant_filter/gsea_report_for_na_neg.xls", sep="\t", header=TRUE)
data[['tumor.variant.neg']]$direction <- -1

merged.data <- ldply(data, function(x) x[, c("NAME", "FDR.q.val", "direction")])

## Process data
merged.data$NAME <- gsub("KEGG_", "", merged.data$NAME)
merged.data$.id <- gsub("\\.pos|\\.neg", "", merged.data$.id)
merged.data$FDR.q.val <- round(merged.data$FDR.q.val, 6)


merged.data$FDR.q.val.formatted <- format.pval(merged.data$FDR.q.val, digits=4)

merged.data$value <- ifelse(merged.data$direction == 1,
                            gsub("(.*)",
                                 '<div style="color: red;">\\1</div>',
                                 merged.data$FDR.q.val),
                            gsub("(.*)", 
                                 '<div style="color: green;">\\1</div>',
                                 merged.data$FDR.q.val))

fdr.thresh <- 0.05
merged.data$value <- ifelse(merged.data$FDR.q.val <= fdr.thresh,
                            gsub("(.*)",
                                 '<div style="font-size:150%">\\1</div>',
                                 merged.data$value),
                            merged.data$value)

  
merged.data.wide <- cast(merged.data, NAME ~ .id, value="value")
rownames(merged.data.wide) <- merged.data.wide$NAME
merged.data.wide <- merged.data.wide[, -1]
merged.data.wide <- as.data.frame(merged.data.wide)

merged.data.wide.fdr <- cast(merged.data, NAME ~ .id, value="FDR.q.val")
rownames(merged.data.wide.fdr) <- rownames(merged.data.wide)
merged.data.wide.fdr <- merged.data.wide.fdr[, -1]

merged.data.wide.fdr <- as.data.frame(apply(merged.data.wide.fdr, 2, 
                                            as.numeric))
rownames(merged.data.wide.fdr) <- rownames(merged.data.wide)
colnames(merged.data.wide.fdr) <- colnames(merged.data.wide)

## Anything that is NA is FDR = 1
merged.data.wide.fdr[is.na(merged.data.wide.fdr)] <- 1

merged.data.wide <- merged.data.wide[order(merged.data.wide.fdr$tumor.variant), ]

merged.data.wide$classic.variant <- as.character(merged.data.wide$classic.variant)
merged.data.wide$tumor.variant <- as.character(merged.data.wide$tumor.variant)
merged.data.wide$tumor.classic <- as.character(merged.data.wide$tumor.classic)

set.na <- is.na(merged.data.wide.fdr)
merged.data.wide[set.na] <- "NS"

merged.data.wide <- merged.data.wide[, c("tumor.variant", "tumor.classic", "classic.variant")]

colnames(merged.data.wide) <- c( "UISO vs. Tumor", "Classic vs. Tumor",
                                 "UISO vs. Classic")

```

## Supplemental Table 1: Gene Set Enrichment Analysis (GSEA) on `r nrow(merged.data.wide)` KEGG pathways using differentially expressed genes in the cell lines

See the following script to reproduce all provided files (assumes that your current directory is the repository root):

```
sh ./data/GSEA/run_GSEA.sh
```

To identify pathways that are affected similarly, we performed three comparisons: UISO cell line vs. tumor samples, classic cell lines vs. tumor samples, and UISO cell line vs. classic cell lines. Green values are up-regulated and red values are down-regulated. Large bold values are statistically significant (FDR-adjusted p-value < `r fdr.thresh`).

```{r TableS1, echo=FALSE, results='asis'} 
tmp <- xtable(merged.data.wide, digits=c(1,2,2,2), align=c("r", "c", "c", "c"))
print(tmp, type='html', sanitize.text.function=I)
``` 

```{r loadccle, echo=FALSE, error=FALSE, warning=FALSE, cache=FALSE, message=FALSE}
load("../cache/rf.eset.nouiso.predict.RData")
load("../cache/rf.eset.nouiso.RData")
```

## Supplemental Table 2: UISO cell line samples are not not classified as MCC with a random forest classifier trained using cell line microarray expression data.

The mean prediction votes over six replicates of UISO cell line samples are given.

```{r TableS2, echo=FALSE, results='asis'}
uisovotes <- as.data.frame(t(colMeans(rf.eset.nouiso.predict)))
rownames(uisovotes) <- "UISO"
print(xtable(uisovotes, digits=2))
```

## Supplemental Table 3: Held out (out-of-bag) WaGa and Mkl-1 cell line samples are classifed as MCC using random forest trained with using cell line microarray expression data.
```{r TableS3, echo=FALSE, results='asis'}
oobvotes <- data.frame(rf.eset.nouiso$votes,
                       cancertype=rf.eset.nouiso$y)

avgoobvotes <- ddply(oobvotes, .variables="cancertype", colwise(mean, is.numeric))

rownames(avgoobvotes) <- avgoobvotes[, 1]
avgoobvotes <- avgoobvotes[, -1]
colnames(avgoobvotes) <- levels(rf.eset.nouiso$y)
print(xtable(avgoobvotes, digits=2))
```

## Supplemental Table 4: Antibodies used in immunohistochemical analysis.

```{r TableS4, results='asis', echo=FALSE}

antibody.data <- read.csv("../data/antibodies.csv")
colnames(antibody.data) <- c("Antibody", "Company", "Dilution", "Antigen retrievial")
antibody.xtbl <- xtable(antibody.data)

print(antibody.xtbl, include.row.names=FALSE, sanitize.text.function=I)
```

## Supplemental Table 5: Short tandem repeat (STR) fingerprinting results for the WaGa, Mkl-1, and UISO cell lines using 16 markers from the AmpFlSTR Identifiler PCR Amplification Kit (Applied Biosystems).

```{r TableS5, results='asis', echo=FALSE}

sky.data <- read.table("../data/sky.tsv", sep="\t", header=TRUE)

colnames(sky.data) <- c("Marker", "UISO (early passage)", "UISO (late passage)", "WaGa", "Mkl-1")

sky.xtbl <- xtable(sky.data)

print(sky.xtbl, sanitize.text.function=I)
```
